src/sys/reboot.rs

```diff
@@ -0,0 +1,44 @@
+use {Errno, Error, Result};
+use libc;
+use void::Void;
+use std::mem::drop;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum RebootMode {
+    Halt,
+    kexec,
+    PowerOff,
+    Restart,
+    // we do not support Restart2,
+    Suspend,
+}
+
+pub fn reboot(how: RebootMode) -> Result<Void> {
+    let cmd = match how {
+        RebootMode::Halt => libc::RB_HALT_SYSTEM,
+        RebootMode::kexec => libc::RB_KEXEC,
+        RebootMode::PowerOff => libc::RB_POWER_OFF,
+        RebootMode::Restart => libc::RB_AUTOBOOT,
+        // we do not support Restart2,
+        RebootMode::Suspend => libc::RB_SW_SUSPEND,
+    };
+    unsafe {
+        libc::reboot(cmd)
+    };
+    Err(Error::Sys(Errno::last()))
```

Comment:
No, the whole point is that `reboot()` never returns, unless there's an error. Same thing as `execvp` and friends.
`set_cad_enabled()` (btw is that a good name?) on the other hand returns if everything went fine.